lib/algos/list/delimiterMaster.js

```diff
@@ -1,84 +1,112 @@
-     * @return {undefined}
+     * Delimiter listing of master versions.
+     * @param {Object} parameters           - listing parameters
+     * @param {String} parameters.delimiter - delimiter per amazon format
+     * @param {String} parameters.prefix    - prefix per amazon format
+     * @param {String} parameters.marker    - marker per amazon format
+     * @param {Number} parameters.maxKeys   - number of keys to list
+     */
+    constructor(parameters) {
+        super(parameters);
+        this.prvPHDKey = undefined;
+    }
+
+    /**
+     *  Add a (key, value) tuple to the listing
+     *  Set the NextMarker to the current key
+     *  Increment the keys counter
+     *  @param {String} key   - The key to add
+     *  @param {String} value - The value of the key
+     *  @return {number}      - indicates if iteration should continue
      */
     addContents(key, value) {
-        this.Contents.push({
-            key,
-            value: {
-                Size: value['content-length'],
-                ETag: value['content-md5'],
-                LastModified: value['last-modified'],
-                // <versioning>
-                VersionId: VSUtils.getts(value),
-                // </versioning>
-                Owner: {
-                    DisplayName: value['owner-display-name'],
-                    ID: value['owner-id'],
-                },
-                StorageClass: value['x-amz-storage-class'],
-                Initiated: value.initiated,
-                Initiator: value.initiator,
-                EventualStorageBucket: value.eventualStorageBucket,
-                partLocations: value.partLocations,
-                creationDate: value.creationDate,
-            },
-        });
+        if (this._reachedMaxKeys()) {
+            return FILTER_END;
+        }
+        this.Contents.push({ key, value });
         this.NextMarker = key;
         ++this.keys;
+        return FILTER_ACCEPT;
     }
 
     /**
-     * Overriding the filter function that formats the
-     * listing results based on the listing algorithm.
-     *
-     * @param {object} obj - metadata entry in the form of { key, value }
-     * @return {boolean} - continue filtering or return the formatted list
+     *  Filter to apply on each iteration, based on:
+     *  - prefix
+     *  - delimiter
+     *  - maxKeys
+     *  The marker is being handled directly by levelDB
+     *  @param {Object} obj       - The key and value of the element
+     *  @param {String} obj.key   - The key of the element
+     *  @param {String} obj.value - The value of the element
+     *  @return {number}         - indicates if iteration should continue
      */
     filter(obj) {
-        // Check first in case of maxkeys <= 0
-        if (this.keys >= this.maxKeys) {
-            // In cases of maxKeys <= 0 => IsTruncated = false
-            this.IsTruncated = this.maxKeys > 0;
-            return false;
+        let key = obj.key;
```

Comment:
No. the derivatives of `Delimiter` are filtering the super-set of what `Delimiter` does.